source/extensions/transport_sockets/tls/context_impl.cc

```diff
@@ -986,33 +989,58 @@ ServerContextImpl::generateHashForSessionContexId(const std::vector<std::string>
     RELEASE_ASSERT(cert != nullptr, "TLS context should have an active certificate");
     X509_NAME* cert_subject = X509_get_subject_name(cert);
     RELEASE_ASSERT(cert_subject != nullptr, "TLS certificate should have a subject");
+
     const int cn_index = X509_NAME_get_index_by_NID(cert_subject, NID_commonName, -1);
-    // It's possible that the certificate doesn't have CommonName, but has SANs.
     if (cn_index >= 0) {
       X509_NAME_ENTRY* cn_entry = X509_NAME_get_entry(cert_subject, cn_index);
       RELEASE_ASSERT(cn_entry != nullptr, "certificate subject CN should be present");
+
       ASN1_STRING* cn_asn1 = X509_NAME_ENTRY_get_data(cn_entry);
-      RELEASE_ASSERT(ASN1_STRING_length(cn_asn1) > 0, "");
+      if (ASN1_STRING_length(cn_asn1) <= 0) {
+        EVP_MD_CTX_cleanup(&md);
```

Comment:
Just change type of `md` to `bssl::ScopedEVP_MD_CTX` so we don't have to worry about cleanup?

Reply:
Fixed.